### PR TITLE
Add command and parser for ToDo, Deadline and Event

### PR DESCRIPTION
**Pull request by dexter-sim**

### CODE REVIEW

File: src/main/java/taskbook/logic/parser/tasks/TaskDeadlineCommandParser.java [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+
+        if (!arePrefixesPresent(argMultimap, firstPrefix, CliSyntax.PREFIX_DESCRIPTION, CliSyntax.PREFIX_DATE)
+                || !argMultimap.getPreamble().isEmpty()) {
+            throw new ParseException(String.format(
+                    Messages.MESSAGE_INVALID_COMMAND_FORMAT, TaskDeadlineCommand.MESSAGE_USAGE));
+        }
+
+        Name name = ParserUtil.parseName(argMultimap.getValue(firstPrefix).get());
+        Description description = ParserUtil.parseDescription(argMultimap.getValue(CliSyntax.PREFIX_DESCRIPTION).get());
+        Assignment assignment =
```
Comment (seetohjinwei) [inline src/main/java/taskbook/logic/parser/tasks/TaskDeadlineCommandParser.java:65]: The command `task add m/Bernice Yu o/Bernice Yu d/Description` in the original `TaskAddCommand` throws the wrong error. Perhaps we should check if both `o/` and `m/` flags are provided and throw a different error instead. I took a quick glance at this function and it does not seem to catch that, unless I missed it out!

```
Names should only contain alphanumeric characters and spaces, and it should not be blank
```

We could do something like this
```java
if (argMultimap.getValue(PREFIX_ASSIGN_TO).isPresent()
    && argMultimap.getValue(PREFIX_ASSIGN_FROM).isPresent()) {
    throw new ParseException(ExceptionGoesHere);
}
```

**Review by seetohjinwei — CHANGES_REQUESTED** [review]
I took a look primarily at the logic bits of the code and found a potential problem!

File: src/main/java/taskbook/logic/parser/tasks/TaskDeadlineCommandParser.java (hunk 1 above)
Comment (dexter-sim) [reply src/main/java/taskbook/logic/parser/tasks/TaskDeadlineCommandParser.java:71]: Good catch.

`task todo m/Bernice Yu o/Bernice Yu d/Description`, as well as the deadline and event variants, will now throw invalid command format message.

**Review by bananamonkey4655 — APPROVED** [review]
LGTM!

**Review by xiaobill8 — APPROVED** [review]
Testcases look good!

**Review by seetohjinwei — APPROVED** [review]

**Pull request merged**